### DISCUSSION TITLE
How to define the number of behavioral variables in the PLS model

### DISCUSSION THREAD

**Discussion by szahar** [McIntosh-Lab/PLS#9] [Q&A]
Hello,

I would like to create a PLS model with 3 groups, 2 conditions and 2 variables. My issue is that I don’t find where to declare the number of behavioral variables (only number of conditions and numbers of subjects per groups are input of the PLS function).

Thank you for your help.

Best,
Sélima

**Comment by randymcintosh** [comment]
Hi Selima – no need to set the number of behavior variables.  The code figures that out

From: szahar ***@***.***>
Date: Tuesday, September 17, 2024 at 11:19 AM
To: McIntosh-Lab/PLS ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [McIntosh-Lab/PLS] How to define the number of behavioral variables in the PLS model (Discussion #9)

Hello,

I would like to create a PLS model with 3 groups, 2 conditions and 2 variables. My issue is that I don’t find where to declare the number of behavioral variables (only number of conditions and numbers of subjects per groups are input of the PLS function).

Thank you for your help.

Best,
Sélima

**Reply by szahar** [reply]
Thanks for the quick reply! 
I was also thinking the function would figure out how many behavioural variables I have. But then the problem is that I end-up with with 6 LVs while the number of LV for Regular Behavior PLS is:  nLVs = nConditions x nGroups x nBehaviorMeasure which should be 2 x 3 x 2=12 in my case. 

Please find below how I initialized my variables: 
```

groups{1}=[P_brain_data];
groups{2}=[C_brain_data];
groups{3}=[A_brain_data];
numconds = 2; 
datamat_lst=cell(length(groups),1);
accuracy=[P_behavior_data';C_behavior_data';A_behavior_data'];

datamat_lst{1} = P_brain_data; 
datamat_lst{2} = C_brain_data; 
datamat_lst{3} = A_brain_data; 


num_subj_lst = [size(P_brain_data,1)/2 size(C_brain_data,1)/2 size(A_brain_data,1)/2];

%% PLS Options
option.method = 3;  
option.num_boot = 10000;
option.num_perm = 10000;
option.meancentering_type = [2]; 
option.stacked_behavdata = [accuracy];
result = pls_analysis(datamat_lst, num_subj_lst, numconds, option);

```

**Comment by randymcintosh** [comment]
Hmm- your syntax looks good, but maybe your behavior vector has one column rather than two?

From: szahar ***@***.***>
Date: Wednesday, September 18, 2024 at 2:22 AM
To: McIntosh-Lab/PLS ***@***.***>
Cc: randymcintosh ***@***.***>, Comment ***@***.***>
Subject: Re: [McIntosh-Lab/PLS] How to define the number of behavioral variables in the PLS model (Discussion #9)

Thanks for the quick reply!
I was also thinking the function would figure out how many behavioural variables I have. But then the problem is that I end-up with with 6 LVs while the number of LV for Regular Behavior PLS is: nLVs = nConditions x nGroups x nBehaviorMeasure which should be 2 x 3 x 2=12 in my case.

Please find below how I initialized my variables:



groups{1}=[P_brain_data];

groups{2}=[C_brain_data];

groups{3}=[A_brain_data];

numconds = 2;

datamat_lst=cell(length(groups),1);

accuracy=[P_behavior_data';C_behavior_data';A_behavior_data'];



datamat_lst{1} = P_brain_data;

datamat_lst{2} = C_brain_data;

datamat_lst{3} = A_brain_data;





num_subj_lst = [size(P_brain_data,1)/2 size(C_brain_data,1)/2 size(A_brain_data,1)/2];



%% PLS Options

option.method = 3;

option.num_boot = 10000;

option.num_perm = 10000;

option.meancentering_type = [2];

option.stacked_behavdata = [accuracy];

result = pls_analysis(datamat_lst, num_subj_lst, numconds, option);

**Comment by szahar** [comment]
Hello, 

Thanks for the reply. 

According to the instructions given in pls.analysis, we need to organize the behavioural matrix for non-rotated behavioural PLS as follow:  
 ```
So total number of rows equal to
%		that multiplied by number of groups, in the form of
%		"measure in condition in group". i.e.:
%			g1	c1	b1
%					b2
%				c2	b1
%					b2
%				c3	b1
%					b2
%			g2	c1	b1
%					b2
%				c2	b1
%					b2
%				c3	b1
%					b2
```

For this reason, I initially assigned each behavioral measure to a row rather than a column. Similarly, I nested conditions and groups in the columns.

Should I instead keep the nesting for groups and conditions but organize the behavioral measures in columns?

Thanks for your reply.

Best,
Sélima
